include/cactus_rt/app.h

```diff
@@ -1,31 +1,31 @@
-#ifndef CACTUS_RT_APP_H_
-#define CACTUS_RT_APP_H_
+#ifndef CACTUS_RT_APP2_H_
+#define CACTUS_RT_APP2_H_
 
 #include <cstddef>
 #include <cstdint>
+#include <map>
+#include <memory>
+
+#include "thread.h"
 
 namespace cactus_rt {
 class App {
-  size_t heap_size_;
-
   /**
-   * The start time of the application from the monotonic clock in nanoseconds.
+   * Size of the heap to reserve in bytes.
    */
-  int64_t start_monotonic_time_ns_;
+  size_t heap_size_;
 
-  /**
-   * The start time of the application from the realtime clock in nanoseconds.
-   * Do not ever use this for any real-time sensitive computations.
-   */
-  int64_t start_wall_time_ns_;
+  // Non-owning references to threads just to help with starting and joining the thrad.
+  std::vector<BaseThread*> threads_;
```

Comment:
Not sure if this is the best idea right now, but it's relatively simple and works. The best alternative AFAICT is a `shared_ptr`, which I'm not sure is an actual improvement. :shrug: